<a href="https://colab.research.google.com/github/Hamza-t/Language-Identification/blob/main/Data_Preparation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#!Hello

* In this notebook, we will merge all data file in one data frame and insure the type of each column 
* The final data will contain just 2 columns : "text" and "label"
* The "text" column shoold be string, the "label" column shoold be integer

* We have 2 datasets : 

1. final_scraped_data (scraped data) or Final_data/all_Data
2. annotated_data (Tunizi/Arabizi)





In [ ]:
#Import library
import pandas as pd

In [ ]:
#Import scraped file 
#drive mount
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
#verify the path
folder_path = "/content/drive/MyDrive/My_work/NLP-Project-INSAT"
!ls "/content/drive/MyDrive/My_work/NLP-Project-INSAT"

'Data augmentation'	  'data collection '	     'General Data'
 Data_Augmentation.ipynb   Data_Preparation.ipynb
 Data_cleaning.ipynb	   Data_visualization.ipynb


In [ ]:
!ls "/content/drive/MyDrive/My_work/NLP-Project-INSAT/General Data"

all_Data.xlsx  scraped_data.xlsx


In [ ]:
#assign data 

all_data = folder_path + "/General Data/" + "all_Data.xlsx"
scraped_data = folder_path +"/General Data/" + "scraped_data.xlsx"

In [ ]:
data00 = pd.read_excel(all_data)
data01 = pd.read_excel(scraped_data)

In [ ]:
data00.head()

text  label
0  ana fil ghorba w sout el athan ahla haja 3ijbi...    3.0
1                           man eheb rasoule mohamed    3.0
2                        rabi yerhmou sofiane cha3ri    3.0
3     rabi yarahmak ya sbou3i wou yarham kol om mita    3.0
4                              merci beaucoup salime    1.0

In [ ]:
data01.head()

COMMENT  Label
0  thank you all for believing in us insat is our...    2.0
1                                i am saying goodbye    2.0
2                         وانا لتوّا ما فهمتش لي صار    0.0
3                  قدامي خليتك هزوك و في ليلة و نهار    0.0
4                 حيوطك الحمرة كالجمرة تكويتي بالنار    0.0

In [ ]:
#change the name of columns in data01
data01 = data01.rename(columns={'COMMENT': 'text', 'Label' : 'label'})
data01.head()

text  label
0  thank you all for believing in us insat is our...    2.0
1                                i am saying goodbye    2.0
2                         وانا لتوّا ما فهمتش لي صار    0.0
3                  قدامي خليتك هزوك و في ليلة و نهار    0.0
4                 حيوطك الحمرة كالجمرة تكويتي بالنار    0.0

In [ ]:
print(f'the size of the first data is {len(data00)}')
print(f'the size of the second data is {len(data01)}')

the size of the first data is 7850
the size of the second data is 486


In [ ]:
#concat the data
data = pd.concat([data00, data01], ignore_index=True)
data 

text  label
0     ana fil ghorba w sout el athan ahla haja 3ijbi...    3.0
1                              man eheb rasoule mohamed    3.0
2                           rabi yerhmou sofiane cha3ri    3.0
3        rabi yarahmak ya sbou3i wou yarham kol om mita    3.0
4                                 merci beaucoup salime    1.0
...                                                 ...    ...
8331  brabi ena maghroum bel info ou behi fel math o...    4.0
8332  ولدي السنة اولى ثانوي اي التوجيه ولدي باهي برش...    0.0
8333                  mercie pour votre conseil chapeau    1.0
8334  section technique andha 4 matieres de base mat...    4.0
8335                       good job bro bac info coding    4.0

[8336 rows x 2 columns]

In [ ]:
print(f'the size of the second data is {len(data)}')

the size of the second data is 8336


* We have 8336 comments in our data, let's clean it! 

In [ ]:
#save data to .csv file
from pathlib import Path  
filepath = Path('/content/drive/MyDrive/My_work/NLP-Project-INSAT/prepared_data.csv') 
data.to_csv(filepath)